suite-common/wallet-utils/src/__tests__/filterReceiveAccounts.test.ts

```diff
@@ -0,0 +1,128 @@
+    getWalletAccount({ symbol: 'eth', accountType: 'ledger' }),
+    getWalletAccount({ symbol: 'btc', accountType: 'coinjoin' }),
+    getWalletAccount({ symbol: 'btc', accountType: 'taproot' }),
+    getWalletAccount({ symbol: 'btc', accountType: 'legacy' }),
+    getWalletAccount({ symbol: 'btc', accountType: 'segwit' }),
+    getWalletAccount({ symbol: 'btc', accountType: 'ledger' }),
+    getWalletAccount({ symbol: 'pol', accountType: 'legacy' }),
+    getWalletAccount({ symbol: 'pol', accountType: 'normal' }),
+    getWalletAccount({ symbol: 'pol', accountType: 'ledger' }),
+    getWalletAccount({ symbol: 'sol', accountType: 'normal', empty: true, visible: false }),
+    getWalletAccount({ symbol: 'sol', accountType: 'ledger' }),
+    getWalletAccount({
+        symbol: 'sol',
+        accountType: 'ledger',
+        empty: true,
+        visible: false,
+    }),
+];
+
+type RunFilterReceiveAccountsTestParams = {
+    isDebug?: boolean;
+    receiveNetwork?: string;
+    deviceState?: `${string}@${string}:${number}`;
+    accounts?: Account[];
+};
+
+const runFilterReceiveAccouns = ({
+    isDebug = true,
+    receiveNetwork = 'eth',
+    deviceState = '1stTestnetAddress@device_id:0',
+    accounts = accountsList,
+}: RunFilterReceiveAccountsTestParams) => {
+    const device = getSuiteDevice({
+        unavailableCapabilities: {
+            dash: 'no-support',
+        },
+        state: deviceState,
+    });
+    const unavailableCapabilities = device?.unavailableCapabilities ?? {};
+
+    const receiveNetworks = networksCollection.filter(
+        (n: Network) =>
+            n.symbol === receiveNetwork &&
+            !unavailableCapabilities[n.symbol] &&
+            ((n.isDebugOnlyNetwork && isDebug) || !n.isDebugOnlyNetwork),
+    );
+
+    return filterReceiveAccounts({
+        accounts,
+        deviceState: device.state,
+        receiveNetwork,
+        isDebug,
+        receiveNetworks,
+    });
+};
+
+describe('filter receive accounts', () => {
+    it('check if account is debug only type', () => {
+        expect(isDebugOnlyAccountType('legacy', 'btc')).toBe(false);
+        expect(isDebugOnlyAccountType('segwit', 'btc')).toBe(false);
+        expect(isDebugOnlyAccountType('coinjoin', 'btc')).toBe(false);
+        expect(isDebugOnlyAccountType('taproot', 'btc')).toBe(false);
+        expect(isDebugOnlyAccountType('ledger', 'btc')).toBe(false);
+        expect(isDebugOnlyAccountType('legacy', 'eth')).toBe(true);
+        expect(isDebugOnlyAccountType('ledger', 'eth')).toBe(true);
+        expect(isDebugOnlyAccountType('normal', 'regtest')).toBe(false);
+    });
+
+    it('empty accounts array', () => {
+        expect(runFilterReceiveAccouns({ accounts: [] })).toEqual([]);
+    });
+
+    it('when given a non-existing network in acccounts list', () => {
+        expect(runFilterReceiveAccouns({ receiveNetwork: 'bnb' })).toEqual([]);
+    });
+
+    it('when debug mode is on', () => {
```

Comment:
```suggestion
    it('filters only debug accounts, when debug mode is on', () => {
```


.. and similarly for rest of the `it`s